scalableminds/webknossos
# Synchronize list focus with active segment

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- create a lot of segments so that one can scroll in the segments view
- click on a segment in the data viewport and ensure that the segments view scrolls and highlights to the corresponding item

### TODOs:
- [x] create context menu entry
- [x] fix issue that only available segments are stored as selected (e.g. after deleting a segment, see commented-out-code). test this by removing a selected segment
- [x] highlight difference between `jump to segment` and `Select segment` in viewport segment dropdown
- [x] make sure the "click segment" action from context menu doesnt harm anything
- [x] actually focus segment in list, so scroll to it via `this.tree.current.scrollTo({ key: selectedElement.key });` with the key being `segment-<id>`
    - [x] fix that scrolling doesnt work for newly added segments 
    -> may be bug
- [x] fix drag and drop in segment list, e.g. to groups (seperate issue: #7527)
- [x] remove logging and read through the rest of the code

### Issues:
- fixes #7311

------
(Please delete unneeded items, merge only when none are left open)
